src/commonMain/kotlin/fr/acinq/eclair/channel/ChannelTypes.kt

```diff
@@ -272,48 +320,51 @@ data class RevokedCommitPublished(
         // even if our txs only have one input, maybe our counterparty uses a different scheme so we need to iterate
         // over all of them to check if they are relevant
         val relevantOutpoints = tx.txIn.map { it.outPoint }.filter { outPoint ->
-            // is this the commit tx itself ? (we could do this outside of the loop...)
+            // is this the commit tx itself? (we could do this outside of the loop...)
             val isCommitTx = commitTx.txid == tx.txid
             // does the tx spend an output of the commitment tx?
             val spendsTheCommitTx = commitTx.txid == outPoint.txid
-            // is the tx one of our 3rd stage delayed txs? (a 3rd stage tx is a tx spending the output of an htlc tx, which
+            // is the tx a 3rd stage txs? (a 3rd stage tx is a tx spending the output of an htlc tx, which
             // is itself spending the output of the commitment tx)
-            val is3rdStageDelayedTx = claimHtlcDelayedPenaltyTxs.map { it.txid }.contains(tx.txid)
-            // does the tx spend an output of an htlc tx? (in which case it may invalidate one of our claim-htlc-delayed-penalty)
-            val spendsHtlcOutput = claimHtlcDelayedPenaltyTxs.flatMap { it.txIn }.map { it.outPoint }.contains(outPoint)
-            isCommitTx || spendsTheCommitTx || is3rdStageDelayedTx || spendsHtlcOutput
+            val is3rdStageDelayedTx = claimHtlcDelayedPenaltyTxs.map { it.input.outPoint }.contains(outPoint)
+            isCommitTx || spendsTheCommitTx || is3rdStageDelayedTx
         }
         // then we add the relevant outpoints to the map keeping track of which txid spends which outpoint
-        return this.copy(irrevocablySpent = irrevocablySpent + relevantOutpoints.map { it to tx.txid }.toMap())
+        return this.copy(irrevocablySpent = irrevocablySpent + relevantOutpoints.map { it to tx }.toMap())
```

Comment:
```suggestion
        return this.copy(irrevocablySpent = irrevocablySpent + relevantOutpoints.associateWith { tx })
```

Reply:
Done in https://github.com/ACINQ/eclair-kmp/pull/229/commits/4b46d6121879bb9ef82542100ae63e5a1fd62ec4